applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py

```diff
@@ -151,29 +147,30 @@ def Initialize(self):
 
         self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DYNAMIC_TAU, self.settings["dynamic_tau"].GetDouble())
 
-        print ("Monolithic embedded solver initialization finished.")
+        KratosMultiphysics.Logger.PrintInfo("NavierStokesEmbeddedMonolithicSolver", "Solver initialization finished.")
 
 
     def InitializeSolutionStep(self):
         (self.bdf_process).Execute()
-        if (self.settings["solver_type"].GetString() == "EmbeddedAusas"):
-            (self.find_nodal_neighbours_process).Execute()
         (self.solver).InitializeSolutionStep()
 
 
+    def SolveSolutionStep(self):
+        # Note that the first two time steps are dropped to fill the BDF buffer
+        if (self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] >= 2):
+            (self.solver).SolveSolutionStep()
+
+
     def Solve(self):
         (self.bdf_process).Execute()
-        if (self.settings["solver_type"].GetString() == "EmbeddedAusas"):
-            (self.find_nodal_neighbours_process).Execute()
-
         # Note that the first two time steps are dropped to fill the BDF buffer
         if (self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] >= 2):
             (self.solver).Solve()
```

Comment:
this one should be implemented as

Predict
InitializeSolutionStep
SolveSolutionStep
FinalizeSolutionStep